From cb30c9b3662dc731a26bff1016e62f239306c34a Mon Sep 17 00:00:00 2001
From: Himanshu Madhani <hmadhani@redhat.com>
Date: Thu, 3 May 2018 23:47:50 -0400
Subject: [PATCH] [scsi] qla2xxx: Remove PCI IDs of deprecated adapter

Message-id: <20180503234750.33648-1-hmadhani@redhat.com>
Patchwork-id: 212501
O-Subject: [RHEL 8 e-stor PATCH v2] qla2xxx: Remove PCI IDs of deprecated adapter
Bugzilla: 1572233
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>
RH-Acked-by: Chris Leech <cleech@redhat.com>
RH-Acked-by: Tom Coughlan <coughlan@redhat.com>

Bugzilla:
=========
https://bugzilla.redhat.com/show_bug.cgi?id=1572233

commit none

Brewbuild:
===========
https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=15977226

Tetsing:
========
N/A. PCI IDs are removed as part of this patch

Signed-off-by: Himanshu Madhani <hmadhani@redhat.com>
Signed-off-by: Herton R. Krzesinski <herton@redhat.com>
---
 drivers/scsi/qla2xxx/qla_os.c | 120 +-----------------------------------------
 1 file changed, 1 insertion(+), 119 deletions(-)

diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index 5c5dcca4..be1f6cb 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -2522,73 +2522,6 @@ qla2x00_set_isp_flags(struct qla_hw_data *ha)
 {
 	ha->device_type = DT_EXTENDED_IDS;
 	switch (ha->pdev->device) {
-	case PCI_DEVICE_ID_QLOGIC_ISP2100:
-		ha->isp_type |= DT_ISP2100;
-		ha->device_type &= ~DT_EXTENDED_IDS;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2100;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP2200:
-		ha->isp_type |= DT_ISP2200;
-		ha->device_type &= ~DT_EXTENDED_IDS;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2100;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP2300:
-		ha->isp_type |= DT_ISP2300;
-		ha->device_type |= DT_ZIO_SUPPORTED;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2300;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP2312:
-		ha->isp_type |= DT_ISP2312;
-		ha->device_type |= DT_ZIO_SUPPORTED;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2300;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP2322:
-		ha->isp_type |= DT_ISP2322;
-		ha->device_type |= DT_ZIO_SUPPORTED;
-		if (ha->pdev->subsystem_vendor == 0x1028 &&
-		    ha->pdev->subsystem_device == 0x0170)
-			ha->device_type |= DT_OEM_001;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2300;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP6312:
-		ha->isp_type |= DT_ISP6312;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2300;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP6322:
-		ha->isp_type |= DT_ISP6322;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2300;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP2422:
-		ha->isp_type |= DT_ISP2422;
-		ha->device_type |= DT_ZIO_SUPPORTED;
-		ha->device_type |= DT_FWI2;
-		ha->device_type |= DT_IIDMA;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2400;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP2432:
-		ha->isp_type |= DT_ISP2432;
-		ha->device_type |= DT_ZIO_SUPPORTED;
-		ha->device_type |= DT_FWI2;
-		ha->device_type |= DT_IIDMA;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2400;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP8432:
-		ha->isp_type |= DT_ISP8432;
-		ha->device_type |= DT_ZIO_SUPPORTED;
-		ha->device_type |= DT_FWI2;
-		ha->device_type |= DT_IIDMA;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2400;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP5422:
-		ha->isp_type |= DT_ISP5422;
-		ha->device_type |= DT_FWI2;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2400;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP5432:
-		ha->isp_type |= DT_ISP5432;
-		ha->device_type |= DT_FWI2;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2400;
-		break;
 	case PCI_DEVICE_ID_QLOGIC_ISP2532:
 		ha->isp_type |= DT_ISP2532;
 		ha->device_type |= DT_ZIO_SUPPORTED;
@@ -2596,29 +2529,6 @@ qla2x00_set_isp_flags(struct qla_hw_data *ha)
 		ha->device_type |= DT_IIDMA;
 		ha->fw_srisc_address = RISC_START_ADDRESS_2400;
 		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP8001:
-		ha->isp_type |= DT_ISP8001;
-		ha->device_type |= DT_ZIO_SUPPORTED;
-		ha->device_type |= DT_FWI2;
-		ha->device_type |= DT_IIDMA;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2400;
-		break;
-	case PCI_DEVICE_ID_QLOGIC_ISP8021:
-		ha->isp_type |= DT_ISP8021;
-		ha->device_type |= DT_ZIO_SUPPORTED;
-		ha->device_type |= DT_FWI2;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2400;
-		/* Initialize 82XX ISP flags */
-		qla82xx_init_flags(ha);
-		break;
-	 case PCI_DEVICE_ID_QLOGIC_ISP8044:
-		ha->isp_type |= DT_ISP8044;
-		ha->device_type |= DT_ZIO_SUPPORTED;
-		ha->device_type |= DT_FWI2;
-		ha->fw_srisc_address = RISC_START_ADDRESS_2400;
-		/* Initialize 82XX ISP flags */
-		qla82xx_init_flags(ha);
-		break;
 	case PCI_DEVICE_ID_QLOGIC_ISP2031:
 		ha->isp_type |= DT_ISP2031;
 		ha->device_type |= DT_ZIO_SUPPORTED;
@@ -2635,9 +2545,6 @@ qla2x00_set_isp_flags(struct qla_hw_data *ha)
 		ha->device_type |= DT_T10_PI;
 		ha->fw_srisc_address = RISC_START_ADDRESS_2400;
 		break;
-	case PCI_DEVICE_ID_QLOGIC_ISPF001:
-		ha->isp_type |= DT_ISPFX00;
-		break;
 	case PCI_DEVICE_ID_QLOGIC_ISP2071:
 		ha->isp_type |= DT_ISP2071;
 		ha->device_type |= DT_ZIO_SUPPORTED;
@@ -2752,18 +2659,9 @@ qla2x00_probe_one(struct pci_dev *pdev, const struct pci_device_id *id)
 
 	bars = pci_select_bars(pdev, IORESOURCE_MEM | IORESOURCE_IO);
 	sht = &qla2xxx_driver_template;
-	if (pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2422 ||
-	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2432 ||
-	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP8432 ||
-	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP5422 ||
-	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP5432 ||
-	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2532 ||
-	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP8001 ||
-	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP8021 ||
+	if (pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2532 ||
 	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2031 ||
 	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP8031 ||
-	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISPF001 ||
-	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP8044 ||
 	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2071 ||
 	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2271 ||
 	    pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2261) {
@@ -6842,25 +6740,9 @@ static const struct pci_error_handlers qla2xxx_err_handler = {
 };
 
 static struct pci_device_id qla2xxx_pci_tbl[] = {
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2100) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2200) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2300) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2312) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2322) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP6312) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP6322) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2422) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2432) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP8432) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP5422) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP5432) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2532) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2031) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP8001) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP8021) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP8031) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISPF001) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP8044) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2071) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2271) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2261) },
-- 
2.9.0

